src/docks/timelinedock.h

```diff
@@ -179,13 +181,15 @@ public slots:
         int selectedTrack;
         bool isMultitrackSelected;
     };
+    QVector<Mlt::Producer> m_producers;
```

Comment:
I will change the name.

Holding reference actually bothered me but I couldn't think of another way to avoid holding reference. You mentioned above that Shotcut uses XML to work around this. How is that done? Could you give me an ideia on where to start?

Reply:
Search the code for "MLT.XML" to see what I mean. The best way to learn a new non-simple codebase is by doing maintenance i.e. bug fixing.